tests/test_bgp_af.rb

```diff
@@ -314,65 +370,59 @@ def test_additional_paths_selection
     #
     # Set and verify
     #
+    if platform == :ios_xr
+      config_ios_xr_dependencies(asn)
+      config('route-policy drop_all', 'end-policy')
+    end
     bgp_af = RouterBgpAF.new(asn, vrf, af)
-    bgp_af.additional_paths_selection = 'drop_all'
-
-    assert_equal(bgp_af.additional_paths_selection, 'drop_all',
-                 'Error: additional-paths selection route-map not set')
 
-    af_string = get_bgp_af_cfg(asn, vrf, af)
-    pattern = /^ *additional-paths selection route-map drop_all$/
+    # Check default value
+    assert_equal(bgp_af.default_additional_paths_selection,
+                 bgp_af.additional_paths_selection)
 
-    assert_match(pattern, af_string,
-                 "Error: 'additional-paths selection route-map drop_all' is " \
-                   'not configured and should be')
-
-    #
-    # Test getter
-    #
-    pattern = /^ *drop_all$/
-    assert_match(pattern, bgp_af.additional_paths_selection,
-                 "Error: 'route-map drop_all' is not configured and should be")
+    # Set value and verify
+    bgp_af.additional_paths_selection = 'drop_all'
+    assert_equal('drop_all', bgp_af.additional_paths_selection,
+                 'Error: additional-paths selection route-map/policy not set')
 
-    #
     # Unset and verify
-    #
-
-    # Do a 'no additional-paths selection route-map drop_all'
     bgp_af.additional_paths_selection =
       bgp_af.default_additional_paths_selection
 
-    af_string = get_bgp_af_cfg(asn, vrf, af)
+    assert_equal(bgp_af.default_additional_paths_selection,
+                 bgp_af.additional_paths_selection)
 
-    refute_match(pattern, af_string,
-                 "Error: 'additional-paths selection route-map drop_all' is " \
-                   'configured and should not be')
+    config('no route-policy drop_all') if platform == :ios_xr
+    bgp_af.destroy
   end
 
   ##
   ## advertise_l2vpn_evpn
   ##
   def advertise_l2vpn_evpn(asn, vrf, af)
+    # Create BGP
+    config_ios_xr_dependencies(asn) if platform == :ios_xr
     bgp_af = RouterBgpAF.new(asn, vrf, af)
 
-    #
-    # Set and verify
-    #
-    val = false
-    bgp_af.advertise_l2vpn_evpn = val
-    assert_equal(val, bgp_af.advertise_l2vpn_evpn,
-                 'Error: advertise l2vpn evpn value does not match set value')
-
-    val = true
-    bgp_af.advertise_l2vpn_evpn = val
-    assert_equal(val, bgp_af.advertise_l2vpn_evpn,
-                 'Error: advertise l2vpn evpn value does not match set value')
-
-    val = bgp_af.default_advertise_l2vpn_evpn
-    bgp_af.advertise_l2vpn_evpn = val
-    assert_equal(val, bgp_af.advertise_l2vpn_evpn,
-                 'Error: advertise l2vpn evpn value does not match default' \
-                 'value')
+    initial = bgp_af.advertise_l2vpn_evpn
+
+    if platform == :ios_xr
+      assert_raises(Cisco::UnsupportedError) do
+        bgp_af.advertise_l2vpn_evpn = !initial
```

Comment:
I was told that setting it to the value it is already set at would cause no commands to occur.  I saw this happen in one of the tests and came up with the !initial as a general answer to the issue.  If I'm mistaken about the no-op, let me know!